src/compiler/checker.ts

```diff
@@ -6112,6 +6135,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 approximateLength: 0,
                 trackedSymbols: undefined,
                 bundled: !!compilerOptions.outFile && !!enclosingDeclaration && isExternalOrCommonJsModule(getSourceFileOfNode(enclosingDeclaration)),
+                isEntityNameVisible: (entityName, shouldComputeAliasToMakeVisible) => isEntityNameVisible(entityName, context.enclosingDeclaration!, shouldComputeAliasToMakeVisible),
+                isExpandoFunctionDeclaration,
+                isOptionalParameter,

```

Comment:
At some point I feel like we need to do something better for this function; it depends on semantic info which I think is not so good for isolatedDeclarations, no? Or is it just here to error when it differs from a syntactic interpretation?

Reply:
> `isOptionalParameter` is also used i the checker and I don't think the syntactic version is adequate i that case.

Yeah, this is the deeper thing I mean; I personally think the checker is way too forgiving here by using the types, when it could use something like the above (plus the IIFE check).